src/rpc/server/scala/src/main/scala/com/ctc/omega_edit/grpc/EditorService.scala

```diff
@@ -203,18 +203,33 @@ class EditorService(implicit val system: ActorSystem) extends Editor {
     Await.result(f, 1.second)
   }
 
+  // Method to get ViewportEventKind based on Int
+  def getViewportEventKind(e: Int): omega_edit.ViewportEventKind = e match {
+    case 1  => omega_edit.ViewportEventKind.VIEWPORT_EVT_CREATE
+    case 2  => omega_edit.ViewportEventKind.VIEWPORT_EVT_EDIT
+    case 4  => omega_edit.ViewportEventKind.VIEWPORT_EVT_UNDO
+    case 8  => omega_edit.ViewportEventKind.VIEWPORT_EVT_CLEAR
+    case 16 => omega_edit.ViewportEventKind.VIEWPORT_EVT_TRANSFORM
+    case 32 => omega_edit.ViewportEventKind.VIEWPORT_EVT_MODIFY
+    case _  => omega_edit.ViewportEventKind.VIEWPORT_EVT_UNDEFINED
+  }
+
   def subscribeToViewportEvents(in: ObjectId): Source[ViewportEvent, NotUsed] =
     in match {
       case Viewport.Id(sid, vid) =>
         val f =
           (editors ? ViewportOp(sid, vid, Viewport.Watch)).mapTo[Result].map {
             case ok: Ok with Viewport.Events =>
               ok.stream
-                .map(u => // TODO: Populate all of the viewport event, not just some of it
+                .map(u =>
                   ViewportEvent(
-                    u.id,
+                    sessionId = u.id,
+                    viewportId = vid,
                     serial = u.change.map(_.id),
-                    data = Option(ByteString.copyFromUtf8(u.data))
+                    data = Option(ByteString.copyFromUtf8(u.data)),
```

Comment:
Oh no, I don't think we want this.

Reply:
So I can edit places where data is being converted to a string, once @arosien 's PR is merged so I can rebase. Or see if there are other places you want that changed and add it to his PR. This would just make it so I am not making the exact same change as he is

Reply:
Sounds good to me.